site/themes/hugo-theme-relearn/layouts/shortcodes/error-codes.html

```diff
@@ -1 +1,17 @@
+{{ $data := index site.Data "errors" }}
 
+{{ range $label, $errors := $data }}
+    <h2>{{ $errors.label }}</h2>
+    {{ range $errorName, $error := $errors }}
+        {{ with $error }}
+            {{ if ne $errorName "label" }}
+            <div class="errorcode">
```

Comment:
We currently use `<h4 id="{code}">{code} - {name}</h4>`.
- It makes each error a clickable headline, so that support can point people at specific errors
- The fragment is set to the code instead of the name, which is shorter and more stable
- The H4 headline level is excluded from "On this page" to keep it short (only the category labels are displayed)

Reply:
The headline level doesn't necessarily need to be hardcoded into the logic, but it was easy to do in the Jekyll toolchain - the client-side JavaScript actually fixes any gaps in headline levels in the "On this page" list (if you have H1, H2, H4, the H4 is treated like an H3). We don't necessarily need to do that in the Hugo toolchain, maybe there is a cleaner solution like passing in the desired headline level to the shortcode (because you know which level it should use in the source file where the manually-written headlines are).

Reply:
Clickable headlines implemented.
As for the automatic headlining level adjustment, this is something that we can work on in a future task